### PR TITLE
fixed verbose overwriting trainer parameters

### LINKED ISSUE #991: `'enable_progress_bar': False` ignored?
**Describe the bug**
`'enable_progress_bar': False` in `pl_trainer_kwargs` does not seem to produce the expected behavior.

**To Reproduce**
```
import numpy as np
import pandas as pd

from darts.dataprocessing.transformers import Scaler
from darts.datasets import AirPassengersDataset
from darts.models import TCNModel

import warnings
warnings.filterwarnings('ignore')

# Read data:
series = AirPassengersDataset().load()
series = series.astype(np.float32)

# Create training and validation sets:
train, val = series.split_after(pd.Timestamp('19590101'))

# Normalize the time series (note: we avoid fitting the transformer on the validation set)
transformer = Scaler()
train_transformed = transformer.fit_transform(train)
val_transformed = transformer.transform(val)
series_transformed = transformer.transform(series)

model = TCNModel(
    pl_trainer_kwargs={
        'enable_progress_bar': False,
    },
    n_epochs=30,
    input_chunk_length=14,
    output_chunk_length=1,
)

model.fit(
    train_transformed,
    val_series=val_transformed,
)
```

**Expected behavior**
I expected to not see the progress bar, which appears anyway.

**System (please complete the following information):**
 - Python version: 3.10.4
 - darts version: 0.19.0

### PR DESCRIPTION
**Pull request by dennisbader**
Fixes #991.

## Summary
- prevents setting `verbose` from overwriting trainer parameters in TorchForecastingModels

### CODE REVIEW

File: darts/models/forecasting/torch_forecasting_model.py [hunk 1]
```diff
@@ -514,17 +514,28 @@ def _init_model(self, trainer: Optional[pl.Trainer] = None) -> None:
...
     ) -> None:
         """Sets up the PyTorch-Lightning trainer for training or prediction."""
-
-        self.trainer_params["enable_model_summary"] = (
-            verbose if self.model.epochs_trained == 0 else False
-        )
-        self.trainer_params["enable_progress_bar"] = verbose
+        trainer_params = {key: val for key, val in self.trainer_params.items()}
+        if verbose is not None:
+            raise_deprecation_warning(
```
Comment (hrzn) [inline darts/models/forecasting/torch_forecasting_model.py:525]: Could this be time to phase out `verbose` in next release maybe?

**Review by hrzn — APPROVED** [review]

File: darts/models/forecasting/torch_forecasting_model.py (hunk 1 above)
Comment (dennisbader) [reply darts/models/forecasting/torch_forecasting_model.py:525]: I thought about this and personally I actually like the flexibility and simplicity of setting verbose at trainnig/inference time. 

What again was the motivation behind removing it?

Comment (hrzn) [reply darts/models/forecasting/torch_forecasting_model.py:525]: Actually you're right, I also kind of like it. Perhaps we could un-deprecate it then? :D

Comment (dennisbader) [reply darts/models/forecasting/torch_forecasting_model.py:525]: Sounds good :)

**Pull request merged**